components/user_settings/display/manage_languages/manage_languages.jsx

```diff
@@ -21,15 +24,57 @@ export default class ManageLanguage extends React.Component {
 
     constructor(props) {
         super(props);
+        const locales = I18n.getLanguages();
+        const userLocale = props.locale;
+        const selectedOption = {value: locales[userLocale].value, label: locales[userLocale].name};
+        this.reactSelect = React.createRef();
+        this.reactSelectContainer = React.createRef();
 
         this.state = {
             locale: props.locale,
+            selectedOption,
             isSaving: false,
         };
     }
 
-    setLanguage = (e) => {
-        this.setState({locale: e.target.value});
+    componentDidMount() {
+        if (this.reactSelectContainer.current) {
+            const reactSelectInput = this.reactSelectContainer.current.querySelector('input');
+            reactSelectInput.addEventListener('keydown', this.handleKeyDown);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.reactSelectContainer.current) {
+            const reactSelectInput = this.reactSelectContainer.current.querySelector('input');
+            reactSelectInput.removeEventListener('keydown', this.handleKeyDown);
+        }
+    }
+
+    handleKeyDown = (e) => {
+        if (isKeyPressed(e, Constants.KeyCodes.ESCAPE) && this.reactSelect.current.state.menuIsOpen) {
+            e.stopPropagation();
+            this.reactSelect.current.onMenuClose();
```

Comment:
`current.state.menuIsOpen` and `current .onMenuClose`, probably works but better if those were not accessed directly.  I suggest to add new state to `ManageLanguage` component, like `openMenu` (boolean) which controls open/close state of ReactSelect.  Changing state could be through container's keydown and ReactSelect's keydown, onMenuClose and onMenuOpen.